torch/csrc/jit/passes/quantization.cpp

```diff
@@ -676,5 +677,257 @@ TORCH_API script::Module InsertObservers(
+
+  quant->addInput(scale_val);
+  quant->addInput(zero_point_val);
+  dequant->addInput(scale_val);
+  dequant->addInput(zero_point_val);
+
+  Value* scalar_type_val = insertScalarType(quant, t);
+  TORCH_INTERNAL_ASSERT(scalar_type_val != nullptr);
+  quant->addInput(scalar_type_val);
+  dequant->addInput(scalar_type_val);
+  return dequant;
+}
+
+std::tuple<std::string, c10::optional<script::Module>> findObserverModule(
+    const script::Module& module, Value* v) {
+  for (const Use& u: v->uses()) {
+    // Note that here we just check for the name of observer, but the ideally
+    // we should be comparing the type of observer, this is a temporary
+    // work around until data only clone of module.clone is supported.
+    auto submodule_name = u.user->inputs().at(0)->debugName();
+    if (u.user->kind() == prim::CallMethod && u.user->s(attr::name) == "forward" &&
+//            u.user->inputs().at(0)->type()->kind() == c10::ClassType &&
+            submodule_name.find("observer_for_") == 0) {
+          auto observer_module = module.find_module(submodule_name);
+          return std::make_tuple(submodule_name, observer_module);
+        }
+      }
+      return std::make_tuple("", c10::nullopt);
+}
+
+IValue getQParam(const script::Module& module, Value* v) {
+    TORCH_INTERNAL_ASSERT(v->type()->isSubtypeOf(TensorType::get()));
+    c10::optional<script::Module> observer_module;
+    std::tie(std::ignore, observer_module) = findObserverModule(module, v);
+    TORCH_INTERNAL_ASSERT(observer_module,
+                          "getQParam expects the corresponding observer for ",
+                          v->debugName(),
+                          " exists.");
+    auto calculate_qparams = observer_module.value().get_method("calculate_qparams");
+    IValue qparams = calculate_qparams(std::vector<IValue>());
+    return qparams;
+}
+
+void quantizeBias(const script::Module& module, Value* v) {
+  // Traverse to the place where this is used
+  Node* aten_call_node = nullptr;
+  std::vector<std::string> ops_with_bias = {"aten::conv2d", "aten::_convolution"};
+  for (const auto& use: v->uses()) {
+    if (std::find(ops_with_bias.begin(), ops_with_bias.end(),
+                  use.user->kind().toQualString()) != ops_with_bias.end()) {
+      // Make sure there is no observer module for bias
+      c10::optional<script::Module> observer_module;
+      std::tie(std::ignore, observer_module) = findObserverModule(module, v);
+      TORCH_INTERNAL_ASSERT(!observer_module,
+                            "bias should not be observed!");
+      Value* activation = use.user->inputs()[0];
+      Value* weight = use.user->inputs()[1];
+      // Get qparam from activation
+      IValue act_qparam = getQParam(module, activation);
+      // Get qparam from weight
+      IValue weight_qparam = getQParam(module, weight);
+      IValue bias_scale =  at::scalar_tensor(
+          c10::Scalar(act_qparam.toTuple()->elements()[0].toTensor().item().toDouble() *
```

Comment:
Should bias scale not be inverse ? 1 /  (a * w). I recollect from this
https://github.com/pytorch/pytorch/commit/a501e7d5be3845aae3b90fc349eff8e0f2fb05af#diff-78aab5b4217aeaed920efab0c5f5f5c8R1490

Reply:
it should be `act_scale * wt_scale` actually:
`res_q = (x_f * w_f) / (x_s * w_s) + b_f / (x_s * w_s)`, it's probably lenet is not very sensitive to numerical errors previously so that we didn't discover this issue for previous implementation.

Reply:
aha got it